pkg/timelock/timelock_test.go

```diff
@@ -22,164 +24,114 @@ func newTestTimelockWorker(
 		},
 		{
-			name: "NewTimelockWorker bad call proxy address provided (fail)",
-			args: args{
-				nodeURL:          testNodeURL,
-				timelockAddress:  testTimelockAddress,
-				callProxyAddress: "0x1234",
-				privateKey:       testPrivateKey,
-				fromBlock:        testFromBlock,
-				pollPeriod:       int64(testPollPeriod),
-				logger:           testLogger,
-			},
-			want:    testWorker,
-			wantErr: true,
+			name:    "failure - bad private key",
+			setup:   func(a *argsT) { a.privateKey = "invalid" },
+			wantErr: "the provided private key is not valid: got invalid",
 		},
 		{
-			name: "NewTimelockWorker bad private key provided (fail)",
-			args: args{
-				nodeURL:          testNodeURL,
-				timelockAddress:  testTimelockAddress,
-				callProxyAddress: testCallProxyAddress,
-				privateKey:       "0123456789",
-				fromBlock:        testFromBlock,
-				pollPeriod:       int64(testPollPeriod),
-				logger:           testLogger,
-			},
-			want:    testWorker,
-			wantErr: true,
+			name:    "failure - bad from block",
+			setup:   func(a *argsT) { a.fromBlock = big.NewInt(-1) },
+			wantErr: "from block can't be a negative number (minimum value 0): got -1",
 		},
 		{
-			name: "NewTimelockWorker bad negative from block provided (fail)",
-			args: args{
-				nodeURL:          testNodeURL,
-				timelockAddress:  testTimelockAddress,
-				callProxyAddress: testCallProxyAddress,
-				privateKey:       testPrivateKey,
-				fromBlock:        big.NewInt(-1),
-				pollPeriod:       int64(testPollPeriod),
-				logger:           testLogger,
-			},
-			want:    testWorker,
-			wantErr: true,
+			name:    "failure - bad poll period",
+			setup:   func(a *argsT) { a.pollPeriod = -1 },
+			wantErr: "poll-period must be a positive non-zero integer: got -1",
 		},
 		{
-			name: "NewTimelockWorker bad poll period provided (fail)",
-			args: args{
-				nodeURL:          testNodeURL,
-				timelockAddress:  testTimelockAddress,
-				callProxyAddress: testCallProxyAddress,
-				privateKey:       testPrivateKey,
-				fromBlock:        testFromBlock,
-				pollPeriod:       0,
-				logger:           testLogger,
-			},
-			want:    testWorker,
-			wantErr: true,
+			name:    "failure - bad event listener poll period",
+			setup:   func(a *argsT) { a.eventListenerPollPeriod = -1 },
+			wantErr: "event-listener-poll-period must be a positive non-zero integer: got -1",
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := NewTimelockWorker(tt.args.nodeURL, tt.args.timelockAddress, tt.args.callProxyAddress,
-				tt.args.privateKey, tt.args.fromBlock, tt.args.pollPeriod, tt.args.eventListenerPollPeriod, tt.args.logger)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("NewTimelockWorker() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
-			if !(reflect.TypeOf(tt.want) == reflect.TypeOf(got)) {
-				t.Errorf("NewTimelockWorker() = %v, want %v", got, tt.want)
+			args := defaultArgs
```

Comment:
might be worth adding t.Parallel()

Reply:
I opted to leave it as is for now. Parallelizing integration tests -- which often rely on a shared data store, such as the geth instance -- can get tricky. It might even work as is, but I didn't explicitly design the tests with parallelism in mind. So I'd rather take the time to think it thoroughly if/when it makes sense.

Reply:
Done.